contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_transform_resize.py

```diff
@@ -0,0 +1,370 @@
+        if not p.get("width") or p.get("width") != self._width_to:
+            logging.error(
+                "expect width is '{}' but got '{}'".format(
+                    self._width, p.get("width")
+                )
+            )
+            is_metadata_good = False
+        if not p.get("height") or p.get("height") != self._height_to:
+            logging.error(
+                "expect height is '{}' but got '{}'".format(
+                    self._height, p.get("height")
+                )
+            )
+            is_metadata_good = False
+
+        if not is_metadata_good:
+            raise SystemError("Error: Checking metadata failed")
+
+    def _extract_metadata_property(self, input: str) -> Dict:
+        properties = {}
+        width_pattern = re.compile(r"Width: (\d+)")
+        height_pattern = re.compile(r"Height: (\d+)")
+
+        # Check if either video or video(image) pattern matches
+        properties["width"] = (
+            width_pattern.search(input).group(1)
+            if width_pattern.search(input)
+            else None
+        )
+        properties["height"] = (
+            height_pattern.search(input).group(1)
+            if height_pattern.search(input)
+            else None
+        )
+        logging.debug("Prperties got from meta: {}".format(properties))
+        return properties
+
+    def delete_file(self):
+        try:
+            if os.path.exists(self._output_file_full_path):
+                os.remove(self._output_file_full_path)
+        except Exception as e:
+            logging.warn(
+                "Error occurred while deleting file: {}".format(str(e))
+            )
+
+
+class GenioProject(BaseHandler):
+    """
+    Genio project manages platforms and codecs, and handles
+    building.
+    Spec: https://download.mediatek.com/aiot/download/release-note/v24.0/v24.0_IoT_Yocto_Feature_Table_v1.0.pdf     # noqa: E501
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._codec_parser_map = {
+            "v4l2h264enc": "h264parse",
+            "v4l2h265enc": "h265parse",
+        }
+
+    def _build_command(self) -> str:
+        """
+        Build the GStreamer commands based on the platform and codec.
+        Returns:
+            str: A GStreamer command based on the platform and
+            codec.
+        """
+        pipeline = (
+            "{} filesrc location={} ! decodebin ! v4l2convert ! "
+            "video/x-raw,width={},height={} ! {} ! {} ! mp4mux ! filesink"
+            " location={}"
+        ).format(
+            GST_LAUNCH_BIN,
+            self._golden_sample,
+            self._width_to,
+            self._height_to,
+            self._codec,
+            self._codec_parser_map.get(self._codec),
+            self._output_file_full_path,
+        )
+        return pipeline
+
+
+def main() -> None:
+    args = register_arguments()
+    p = project_factory(**vars(args))
```

Comment:
If you apply the change on line 113 you can just pass in args instead of calling `vars` and unpacking it